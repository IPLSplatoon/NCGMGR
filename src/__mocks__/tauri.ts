export const mockTauriEvent = {
    listen: jest.fn()
}

jest.mock('@tauri-apps/api/event', () => mockTauriEvent)

export const mockTauri = {
    invoke: jest.fn()
}

jest.mock('@tauri-apps/api/tauri', () => mockTauri)

export const mockTauriDialog = {
    open: jest.fn()
}

jest.mock('@tauri-apps/api/dialog', () => mockTauriDialog)

export const mockTauriFs = {
    readDir: jest.fn(),
    readTextFile: jest.fn(),
    removeDir: jest.fn(),
    removeFile: jest.fn(),
    createDir: jest.fn(),
    writeFile: jest.fn()
}

jest.mock('@tauri-apps/api/fs', () => mockTauriFs)

export const mockTauriOs = {
    type: jest.fn()
}

jest.mock('@tauri-apps/api/os', () => mockTauriOs)

export const mockTauriShell = {
    open: jest.fn()
}

jest.mock('@tauri-apps/api/shell', () => mockTauriShell)
