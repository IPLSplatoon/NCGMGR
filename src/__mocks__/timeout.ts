export function mockTimeout (): void {
    jest.spyOn(window, 'setTimeout').mockImplementation(handler => {
        // noinspection SuspiciousTypeOfGuard
        if (typeof handler !== 'string') {
            handler()
        }
        return 0
    })
}
