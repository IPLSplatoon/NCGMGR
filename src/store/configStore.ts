import { defineStore } from 'pinia'

export const LOCAL_STORAGE_CONFIG_KEY = 'config'

export interface Configuration {
    installPath: string
}

export const useConfigStore = defineStore('config', {
    state: () => ({
        installPath: ''
    } as Configuration),
    actions: {
        load () {
            const config = JSON.parse(localStorage.getItem(LOCAL_STORAGE_CONFIG_KEY) ?? '{}')
            this.$patch(config)
        },
        persist () {
            localStorage.setItem(LOCAL_STORAGE_CONFIG_KEY, JSON.stringify(this.$state))
        }
    }
})
