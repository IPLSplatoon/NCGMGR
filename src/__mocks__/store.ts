import { createStore, Store } from 'vuex'
import { Configuration } from '@/store/config'
import { LogStore } from '@/store/log'

export function createConfigStore (): Store<Configuration> {
    return createStore<Configuration>({
        state: {
            installPath: '/install/path'
        },
        actions: {
            load: jest.fn(),
            persist: jest.fn()
        },
        mutations: {
            setInstallPath: jest.fn()
        }
    })
}

export function createLogStore (): Store<LogStore> {
    return createStore<LogStore>({
        state: {
            lines: [],
            completed: false
        },
        mutations: {
            reset: jest.fn(),
            insertLine: jest.fn(),
            setCompleted: jest.fn()
        }
    })
}
