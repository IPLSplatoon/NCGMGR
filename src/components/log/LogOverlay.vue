<template>
    <mgr-overlay v-model:visible="modelVisible" class="log-overlay" :no-background-close="noBackgroundClose" max-width>
        <div class="layout vertical center-horizontal">
            <h2 class="m-b-8">{{title}}</h2>
            <progress-display
                v-if="withProgress"
                :log-key="logKey"
                class="m-b-8 max-width"
            />
            <log-display :log-key="logKey" class="m-b-8" />
            <ipl-button @click="modelVisible = false" :disabled="!completed" data-test="close-button" label="Close" />
        </div>
    </mgr-overlay>
</template>

<script lang="ts">
import { computed, defineComponent } from 'vue'
import MgrOverlay from '@/components/mgr/MgrOverlay.vue'
import { IplButton } from '@iplsplatoon/vue-components'
import { useLogStore } from '@/store/logStore'
import LogDisplay from '@/components/log/LogDisplay.vue'
import ProgressDisplay from '@/components/log/ProgressDisplay.vue'
import { ActionState } from '@/types/log'

export default defineComponent({
    name: 'LogOverlay',

    emits: ['update:visible'],

    components: { ProgressDisplay, LogDisplay, MgrOverlay, IplButton },

    props: {
        title: {
            type: String,
            required: true
        },
        visible: {
            type: Boolean,
            required: true
        },
        logKey: {
            type: String,
            required: true
        },
        noBackgroundClose: {
            type: Boolean,
            default: false
        },
        withProgress: {
            type: Boolean,
            default: false
        }
    },

    setup (props, { emit }) {
        const logStore = useLogStore()

        return {
            completed: computed(() => logStore.actionStates[props.logKey] !== ActionState.INCOMPLETE),
            modelVisible: computed({
                get () {
                    return props.visible
                },
                set (newValue: boolean) {
                    emit('update:visible', newValue)
                }
            })
        }
    }
})
</script>
