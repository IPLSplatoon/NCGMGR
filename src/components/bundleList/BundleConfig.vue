<template>
    <div
        class="layout horizontal"
        style="padding: 8px 0;"
    >
        <ipl-space class="max-width h-max-content">
            <ipl-message
                v-if="versionsError"
                type="error"
                class="m-b-8"
            >
                {{ versionsError }}
            </ipl-message>
            <ipl-select
                v-model="selectedVersion"
                :disabled="disableVersionChange"
                :options="versionOptions"
                label="Version"
            />
            <ipl-button
                label="Change version"
                :disabled="disableVersionChange || bundle.version === selectedVersion"
                class="m-t-8"
                @click="setVersion"
            />
            <log-overlay
                v-model:visible="showInstallLog"
                title="Installing..."
                log-key="change-bundle-version"
            />
        </ipl-space>
        <ipl-space class="max-width m-l-8 h-max-content">
            <ipl-button
                label="Open folder"
                @click="openBundleFolder"
            />
            <ipl-button
                label="Open in terminal"
                :disabled="!configStore.allowOpenInTerminal"
                class="m-t-8"
                @click="openBundleInTerminal"
            />
            <ipl-button
                :label="hasConfigFile ? 'Open config file' : 'Create config file'"
                :color="hasConfigFile ? 'blue' : 'green'"
                :disabled="configFileLoading"
                class="m-t-8"
                @click="openOrCreateConfigFile"
            />
        </ipl-space>
    </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import { computed, PropType, ref, watch } from 'vue'
import { Bundle, configFileExists, createConfigFile, getBundleVersions, openConfigFile } from '@/service/nodecgService'
import { IplButton, IplMessage, IplSelect, IplSpace } from '@iplsplatoon/vue-components'
import { useConfigStore } from '@/store/configStore'
import LogOverlay from '@/components/log/LogOverlay.vue'
import { useLogStore } from '@/store/logStore'
import { invoke } from '@tauri-apps/api/core'
import { useNodecgStore } from '@/store/nodecgStore'
import { open } from '@tauri-apps/plugin-shell'

export default defineComponent({
    name: 'BundleConfig',

    components: { IplMessage, IplButton, IplSelect, IplSpace, LogOverlay },

    props: {
        bundle: {
            type: Object as PropType<Bundle>,
            required: true
        }
    },

    setup (props) {
        const logStore = useLogStore()
        const configStore = useConfigStore()
        const nodecgStore = useNodecgStore()

        const versions = ref<string[]>([])
        const versionsError = ref<string | null>(null)
        const versionsLoading = ref(true)
        const selectedVersion = ref<string | undefined>('')
        const showInstallLog = ref(false)
        const hasConfigFile = ref(false)
        const configFileLoading = ref(true)

        watch(showInstallLog, newValue => {
            if (!newValue) {
                nodecgStore.getBundleList()
            }
        })

        watch(() => props.bundle.name, newValue => {
            getBundleVersions(newValue).then(result => {
                versions.value = result
                if (result.length > 0) {
                    selectedVersion.value = props.bundle.version
                }
            }).catch(e => {
                versionsError.value = String(e)
            }).finally(() => {
                versionsLoading.value = false
            })

            checkConfigFile(newValue)
        }, { immediate: true })

        async function checkConfigFile (bundleName: string): Promise<void> {
            configFileLoading.value = true
            hasConfigFile.value = false
            configFileExists(bundleName, configStore.userConfig.nodecgInstallDir).then(result => {
                hasConfigFile.value = result
            }).catch(e => {
                hasConfigFile.value = false
                console.error(e)
            }).finally(() => {
                configFileLoading.value = false
            })
        }

        function getBundlePath () {
            return `${configStore.userConfig.nodecgInstallDir}/bundles/${props.bundle.name}`
        }

        return {
            configStore,
            hasConfigFile,
            configFileLoading,
            showInstallLog,
            selectedVersion,
            versionsLoading,
            versionsError,
            versions,
            disableVersionChange: computed(() => versionsLoading.value || !!versionsError.value || versions.value.length <= 0),
            versionOptions: computed(() => {
                if (versionsLoading.value) {
                    return [{ name: 'Loading...', value: '' }]
                } else if (versions.value.length <= 0) {
                    return [{ name: 'No versions found', value: '' }]
                } else {
                    return versions.value.map(version => ({ name: version, value: version }))
                }
            }),
            async setVersion () {
                const logKey = 'change-bundle-version'
                logStore.reset(logKey)
                await logStore.listen(logKey, true)
                showInstallLog.value = true
                const invocation = invoke('set_bundle_version', {
                    bundleName: props.bundle.name,
                    version: selectedVersion.value
                })
                logStore.logPromiseResult({ promise: invocation, key: logKey })
                logStore.listenForProcessExit({ key: logKey })
            },
            async openBundleFolder () {
                await open(getBundlePath())
            },
            async openOrCreateConfigFile () {
                if (!hasConfigFile.value) {
                    await createConfigFile(props.bundle.name, configStore.userConfig.nodecgInstallDir)
                    checkConfigFile(props.bundle.name)
                }

                await openConfigFile(props.bundle.name, configStore.userConfig.nodecgInstallDir)
            },
            async openBundleInTerminal () {
                await invoke('open_path_in_terminal', { path: getBundlePath() })
            }
        }
    }
})
</script>
