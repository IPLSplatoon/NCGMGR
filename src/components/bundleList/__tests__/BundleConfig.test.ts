import { createTestingPinia, TestingPinia } from '@pinia/testing'
import { config, flushPromises, mount } from '@vue/test-utils'
import { getBundleVersions, configFileExists, openConfigFile, createConfigFile } from '@/service/nodecgService'
import BundleConfig from '../BundleConfig.vue'
import Mock = jest.Mock
import type { IplSelect, IplButton } from '@iplsplatoon/vue-components'
import { useLogStore } from '@/store/logStore'
import { mockTauri, mockTauriOs, mockTauriShell } from '@/__mocks__/tauri'
import { useConfigStore } from '@/store/configStore'
import LogOverlay from '@/components/log/LogOverlay.vue'

jest.mock('@/service/nodecgService')

describe('BundleConfig', () => {
    let pinia: TestingPinia
    config.global.stubs = {
        IplButton: true,
        IplSelect: true,
        FontAwesomeIcon: true,
        LogOverlay: true
    }

    beforeEach(() => {
        pinia = createTestingPinia()
        config.global.plugins = [pinia];

        (getBundleVersions as Mock).mockResolvedValue(['1.0.0', '0.0.1']);
        (configFileExists as Mock).mockResolvedValue(true)
        mockTauriOs.type.mockResolvedValue('')
    })

    it('matches snapshot', () => {
        const wrapper = mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'bundle-name',
                    version: '0.0.1'
                }
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
    })

    it('matches snapshot after bundle versions are loaded', async () => {
        const wrapper = mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'bundle-name',
                    version: '0.0.1'
                }
            }
        })
        await flushPromises()

        expect(wrapper.html()).toMatchSnapshot()
    })

    it('matches snapshot when no bundle versions are found', async () => {
        (getBundleVersions as Mock).mockResolvedValue([])
        const wrapper = mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'bundle-name',
                    version: '0.0.1'
                }
            }
        })
        await flushPromises()

        expect(wrapper.html()).toMatchSnapshot()
    })

    it('matches snapshot when config file exists', async () => {
        (configFileExists as Mock).mockResolvedValue(true)
        const wrapper = mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'bundle-name',
                    version: '0.0.1'
                }
            }
        })
        await flushPromises()

        expect(wrapper.html()).toMatchSnapshot()
    })

    it('matches snapshot when config file is missing', async () => {
        (configFileExists as Mock).mockResolvedValue(false)
        const wrapper = mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'bundle-name',
                    version: '0.0.1'
                }
            }
        })
        await flushPromises()

        expect(wrapper.html()).toMatchSnapshot()
    })

    it('matches snapshot when getting bundle versions fails', async () => {
        (getBundleVersions as Mock).mockRejectedValue(new Error('Error!'))
        const wrapper = mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'bundle-name',
                    version: '0.0.1'
                }
            }
        })
        await flushPromises()

        expect(wrapper.html()).toMatchSnapshot()
    })

    it('adds expected version options when loading', () => {
        const wrapper = mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'bundle-name',
                    version: '0.0.1'
                }
            }
        })

        expect(wrapper.getComponent<typeof IplSelect>('[data-test="version-selector"]').vm.$props.options)
            .toEqual([{
                name: 'Loading...',
                value: ''
            }])
    })

    it('adds expected version options when no versions are present', async () => {
        (getBundleVersions as Mock).mockResolvedValue([])
        const wrapper = mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'bundle-name',
                    version: '0.0.1'
                }
            }
        })
        await flushPromises()

        expect(wrapper.getComponent<typeof IplSelect>('[data-test="version-selector"]').vm.$props.options)
            .toEqual([{
                name: 'No versions found',
                value: ''
            }])
    })

    it('adds expected version options when version list is present', async () => {
        const wrapper = mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'bundle-name',
                    version: '0.0.1'
                }
            }
        })
        await flushPromises()

        expect(wrapper.getComponent<typeof IplSelect>('[data-test="version-selector"]').vm.$props.options)
            .toEqual([
                {
                    name: '1.0.0',
                    value: '1.0.0'
                },
                {
                    name: '0.0.1',
                    value: '0.0.1'
                }
            ])
    })

    it('handles setting version', async () => {
        mockTauri.invoke.mockResolvedValue({})
        useConfigStore().installPath = 'nodecg/path'
        const logStore = useLogStore()
        logStore.reset = jest.fn()
        logStore.listen = jest.fn()
        logStore.logPromiseResult = jest.fn()
        const wrapper = mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'bundle-name',
                    version: '0.0.1'
                }
            }
        })
        await flushPromises()

        wrapper.getComponent<typeof IplSelect>('[data-test="version-selector"]').vm.$emit('update:modelValue', '2.0.1')
        wrapper.getComponent<typeof IplButton>('[data-test="set-version-button"]').vm.$emit('click')
        await flushPromises()

        expect(logStore.reset).toHaveBeenCalledWith('change-bundle-version')
        expect(logStore.listen).toHaveBeenCalledWith('change-bundle-version', true)
        expect(mockTauri.invoke).toHaveBeenCalledWith('set_bundle_version', {
            bundleName: 'bundle-name',
            version: '2.0.1',
            nodecgPath: 'nodecg/path'
        })
        expect(logStore.logPromiseResult).toHaveBeenCalledWith({
            promise: expect.anything(),
            key: 'change-bundle-version'
        })
        expect(wrapper.getComponent<typeof LogOverlay>('[data-test="bundle-log-overlay"]').vm.$props.visible)
            .toEqual(true)
    })

    it('handles opening bundle folder', () => {
        useConfigStore().installPath = 'nodecg/path'
        const wrapper = mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'bundle-name',
                    version: '0.0.1'
                }
            }
        })

        wrapper.getComponent<typeof IplButton>('[data-test="open-folder-button"]').vm.$emit('click')

        expect(mockTauriShell.open).toHaveBeenCalledWith('nodecg/path/bundles/bundle-name')
    })

    it('disables opening bundle in terminal if system is linux', async () => {
        mockTauriOs.type.mockResolvedValue('linux')
        const wrapper = mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'bundle-name',
                    version: '0.0.1'
                }
            }
        })
        await flushPromises()

        expect(wrapper.getComponent<typeof IplButton>('[data-test="open-in-terminal-button"]').vm.$props.disabled).toEqual(true)
    })

    it.each(['windows', 'macos'])('enables opening bundle in terminal if system is %s', async (type) => {
        mockTauriOs.type.mockResolvedValue(type)
        const wrapper = mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'bundle-name',
                    version: '0.0.1'
                }
            }
        })
        await flushPromises()

        expect(wrapper.getComponent<typeof IplButton>('[data-test="open-in-terminal-button"]').vm.$props.disabled).toEqual(false)
    })

    it('handles opening bundle in terminal', () => {
        useConfigStore().installPath = '/nodecg/install/path'
        const wrapper = mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'cool-bundle',
                    version: '0.0.1'
                }
            }
        })

        wrapper.getComponent<typeof IplButton>('[data-test="open-in-terminal-button"]').vm.$emit('click')

        expect(mockTauri.invoke).toHaveBeenCalledWith(
            'open_path_in_terminal',
            { path: '/nodecg/install/path/bundles/cool-bundle' })
    })

    it('disables opening config file if config file is missing', async () => {
        (configFileExists as Mock).mockResolvedValue(false)
        useConfigStore().installPath = '/nodecg/install/path'
        const wrapper = await mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'cool-bundle',
                    version: '0.0.1'
                }
            }
        })

        expect(configFileExists).toHaveBeenCalledWith('cool-bundle', '/nodecg/install/path')
        expect(wrapper.getComponent<typeof IplButton>('[data-test="open-config-file-button"]').vm.$props.disabled).toEqual(true)
    })

    it('disables opening config file if checking for config file presence fails', async () => {
        (configFileExists as Mock).mockRejectedValue(new Error('Error!'))
        useConfigStore().installPath = '/nodecg/install/path'
        const wrapper = await mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'cool-bundle',
                    version: '0.0.1'
                }
            }
        })

        expect(configFileExists).toHaveBeenCalledWith('cool-bundle', '/nodecg/install/path')
        expect(wrapper.getComponent<typeof IplButton>('[data-test="open-config-file-button"]').vm.$props.disabled).toEqual(true)
    })

    it('enables opening config file if config file exists', async () => {
        (configFileExists as Mock).mockResolvedValue(true)
        useConfigStore().installPath = '/nodecg/install'
        const wrapper = mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'rad-bundle',
                    version: '0.0.1'
                }
            }
        })
        await flushPromises()

        expect(configFileExists).toHaveBeenCalledWith('rad-bundle', '/nodecg/install')
        expect(wrapper.getComponent<typeof IplButton>('[data-test="open-config-file-button"]').vm.$props.disabled).toEqual(false)
    })

    it('handles opening config file if it exists', async () => {
        (configFileExists as Mock).mockResolvedValue(true)
        useConfigStore().installPath = '/nodecg/install'
        const wrapper = await mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'rad-bundle',
                    version: '0.0.1'
                }
            }
        })

        wrapper.getComponent<typeof IplButton>('[data-test="open-config-file-button"]').vm.$emit('click')

        expect(configFileExists).toHaveBeenCalledWith('rad-bundle', '/nodecg/install')
        expect(openConfigFile).toHaveBeenCalledWith('rad-bundle', '/nodecg/install')
    })

    it('handles creating config file if it is missing', async () => {
        (configFileExists as Mock).mockResolvedValue(false);
        (createConfigFile as Mock).mockResolvedValue({})
        useConfigStore().installPath = '/nodecg/path'
        const wrapper = await mount(BundleConfig, {
            props: {
                bundle: {
                    name: 'rad-bundle',
                    version: '0.0.1'
                }
            }
        })

        await wrapper.getComponent<typeof IplButton>('[data-test="open-config-file-button"]').vm.$emit('click')
        await flushPromises()

        expect(configFileExists).toHaveBeenCalledTimes(2)
        expect(configFileExists).toHaveBeenCalledWith('rad-bundle', '/nodecg/path')
        expect(createConfigFile).toHaveBeenCalledWith('rad-bundle', '/nodecg/path')
        expect(openConfigFile).toHaveBeenCalledWith('rad-bundle', '/nodecg/path')
    })
})
