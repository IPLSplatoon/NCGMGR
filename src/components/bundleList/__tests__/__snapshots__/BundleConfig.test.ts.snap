// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BundleConfig matches snapshot 1`] = `
<div class="ipl-space color-light layout horizontal m-t-8 m-b-8" data-v-11cc5f1c="">
  <div class="ipl-space color-dark max-width h-max-content" data-v-11cc5f1c="">
    <!--v-if-->
    <ipl-select-stub modelvalue="" label="Version" options="[object Object]" disabled="true" data-test="version-selector"></ipl-select-stub>
    <ipl-button-stub label="Change version" color="blue" disabled="true" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="set-version-button"></ipl-button-stub>
    <log-overlay-stub title="Installing..." logkey="change-bundle-version" visible="false" nobackgroundclose="true" withprogress="true" data-test="bundle-log-overlay"></log-overlay-stub>
  </div>
  <div class="ipl-space color-dark max-width m-l-8 h-max-content" data-v-11cc5f1c="">
    <ipl-button-stub label="Open folder" color="blue" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" data-test="open-folder-button"></ipl-button-stub>
    <ipl-button-stub label="Open in terminal" color="blue" disabled="true" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="open-in-terminal-button"></ipl-button-stub>
    <ipl-button-stub label="Create config file" color="green" disabled="true" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="open-config-file-button"></ipl-button-stub>
  </div>
</div>
`;

exports[`BundleConfig matches snapshot after bundle versions are loaded 1`] = `
<div class="ipl-space color-light layout horizontal m-t-8 m-b-8" data-v-11cc5f1c="">
  <div class="ipl-space color-dark max-width h-max-content" data-v-11cc5f1c="">
    <!--v-if-->
    <ipl-select-stub modelvalue="0.0.1" label="Version" options="[object Object],[object Object]" disabled="false" data-test="version-selector"></ipl-select-stub>
    <ipl-button-stub label="Change version" color="blue" disabled="true" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="set-version-button"></ipl-button-stub>
    <log-overlay-stub title="Installing..." logkey="change-bundle-version" visible="false" nobackgroundclose="true" withprogress="true" data-test="bundle-log-overlay"></log-overlay-stub>
  </div>
  <div class="ipl-space color-dark max-width m-l-8 h-max-content" data-v-11cc5f1c="">
    <ipl-button-stub label="Open folder" color="blue" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" data-test="open-folder-button"></ipl-button-stub>
    <ipl-button-stub label="Open in terminal" color="blue" disabled="true" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="open-in-terminal-button"></ipl-button-stub>
    <ipl-button-stub label="Open config file" color="blue" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="open-config-file-button"></ipl-button-stub>
  </div>
</div>
`;

exports[`BundleConfig matches snapshot when config file exists 1`] = `
<div class="ipl-space color-light layout horizontal m-t-8 m-b-8" data-v-11cc5f1c="">
  <div class="ipl-space color-dark max-width h-max-content" data-v-11cc5f1c="">
    <!--v-if-->
    <ipl-select-stub modelvalue="0.0.1" label="Version" options="[object Object],[object Object]" disabled="false" data-test="version-selector"></ipl-select-stub>
    <ipl-button-stub label="Change version" color="blue" disabled="true" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="set-version-button"></ipl-button-stub>
    <log-overlay-stub title="Installing..." logkey="change-bundle-version" visible="false" nobackgroundclose="true" withprogress="true" data-test="bundle-log-overlay"></log-overlay-stub>
  </div>
  <div class="ipl-space color-dark max-width m-l-8 h-max-content" data-v-11cc5f1c="">
    <ipl-button-stub label="Open folder" color="blue" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" data-test="open-folder-button"></ipl-button-stub>
    <ipl-button-stub label="Open in terminal" color="blue" disabled="true" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="open-in-terminal-button"></ipl-button-stub>
    <ipl-button-stub label="Open config file" color="blue" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="open-config-file-button"></ipl-button-stub>
  </div>
</div>
`;

exports[`BundleConfig matches snapshot when config file is missing 1`] = `
<div class="ipl-space color-light layout horizontal m-t-8 m-b-8" data-v-11cc5f1c="">
  <div class="ipl-space color-dark max-width h-max-content" data-v-11cc5f1c="">
    <!--v-if-->
    <ipl-select-stub modelvalue="0.0.1" label="Version" options="[object Object],[object Object]" disabled="false" data-test="version-selector"></ipl-select-stub>
    <ipl-button-stub label="Change version" color="blue" disabled="true" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="set-version-button"></ipl-button-stub>
    <log-overlay-stub title="Installing..." logkey="change-bundle-version" visible="false" nobackgroundclose="true" withprogress="true" data-test="bundle-log-overlay"></log-overlay-stub>
  </div>
  <div class="ipl-space color-dark max-width m-l-8 h-max-content" data-v-11cc5f1c="">
    <ipl-button-stub label="Open folder" color="blue" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" data-test="open-folder-button"></ipl-button-stub>
    <ipl-button-stub label="Open in terminal" color="blue" disabled="true" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="open-in-terminal-button"></ipl-button-stub>
    <ipl-button-stub label="Create config file" color="green" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="open-config-file-button"></ipl-button-stub>
  </div>
</div>
`;

exports[`BundleConfig matches snapshot when getting bundle versions fails 1`] = `
<div class="ipl-space color-light layout horizontal m-t-8 m-b-8" data-v-11cc5f1c="">
  <div class="ipl-space color-dark max-width h-max-content" data-v-11cc5f1c="">
    <div class="ipl-message__wrapper ipl-message__type-error m-b-8" data-v-ec3b4444="">
      <font-awesome-icon-stub icon="exclamation-circle" border="false" fixedwidth="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" class="icon" data-v-ec3b4444=""></font-awesome-icon-stub>
      <div class="message-content" data-v-ec3b4444="">
        <!---->Error: Error!
      </div>
    </div>
    <ipl-select-stub modelvalue="" label="Version" options="[object Object]" disabled="true" data-test="version-selector"></ipl-select-stub>
    <ipl-button-stub label="Change version" color="blue" disabled="true" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="set-version-button"></ipl-button-stub>
    <log-overlay-stub title="Installing..." logkey="change-bundle-version" visible="false" nobackgroundclose="true" withprogress="true" data-test="bundle-log-overlay"></log-overlay-stub>
  </div>
  <div class="ipl-space color-dark max-width m-l-8 h-max-content" data-v-11cc5f1c="">
    <ipl-button-stub label="Open folder" color="blue" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" data-test="open-folder-button"></ipl-button-stub>
    <ipl-button-stub label="Open in terminal" color="blue" disabled="true" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="open-in-terminal-button"></ipl-button-stub>
    <ipl-button-stub label="Open config file" color="blue" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="open-config-file-button"></ipl-button-stub>
  </div>
</div>
`;

exports[`BundleConfig matches snapshot when no bundle versions are found 1`] = `
<div class="ipl-space color-light layout horizontal m-t-8 m-b-8" data-v-11cc5f1c="">
  <div class="ipl-space color-dark max-width h-max-content" data-v-11cc5f1c="">
    <!--v-if-->
    <ipl-select-stub modelvalue="" label="Version" options="[object Object]" disabled="true" data-test="version-selector"></ipl-select-stub>
    <ipl-button-stub label="Change version" color="blue" disabled="true" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="set-version-button"></ipl-button-stub>
    <log-overlay-stub title="Installing..." logkey="change-bundle-version" visible="false" nobackgroundclose="true" withprogress="true" data-test="bundle-log-overlay"></log-overlay-stub>
  </div>
  <div class="ipl-space color-dark max-width m-l-8 h-max-content" data-v-11cc5f1c="">
    <ipl-button-stub label="Open folder" color="blue" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" data-test="open-folder-button"></ipl-button-stub>
    <ipl-button-stub label="Open in terminal" color="blue" disabled="true" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="open-in-terminal-button"></ipl-button-stub>
    <ipl-button-stub label="Open config file" color="blue" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="open-config-file-button"></ipl-button-stub>
  </div>
</div>
`;
